tests/capture/transforms/test_capture_single_qubit_fusion.py

```diff
@@ -0,0 +1,699 @@
+
+        jaxpr = jax.make_jaxpr(circuit)(np.pi)
+        assert len(jaxpr.eqns) == 3
+
+        assert jaxpr.eqns[0].primitive == qml.CNOT._primitive
+        assert jaxpr.eqns[1].primitive == for_loop_prim
+        assert jaxpr.eqns[2].primitive == qml.CNOT._primitive
+
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, np.pi)
+        jaxpr_ops = collector.state["ops"]
+        assert len(jaxpr_ops) == 4
+
+        expected_ops = [
+            qml.CNOT(wires=[0, 1]),
+            qml.Rot(np.pi / 2, np.pi, np.pi / 2, wires=[0]),  # RX and RZ fused
+            qml.Hadamard(wires=[0]),  # Should not be fused because is returned
+            qml.CNOT(wires=[0, 1]),
+        ]
+
+        for op1, op2 in zip(jaxpr_ops, expected_ops):
+            # The qml.equal function does not recognize two qml.Rot operators
+            # as equivalent unless the input angles are exactly the same
+            assert op1.name == op2.name
+            assert qml.math.allclose(op1.parameters, op2.parameters)
+            assert op1.wires == op2.wires
+
+    def test_while_loop(self):
+        """Test that while operators inside a while loop are correctly fused."""
+
+        @SingleQubitFusionInterpreter()
+        def circuit():
+
+            qml.CNOT(wires=[0, 1])
+
+            def while_f(i):
+                return i < 3
+
+            @qml.while_loop(while_f)
+            def loop(i):
+                qml.RX(np.pi, wires=0)
+                qml.RZ(np.pi, wires=0)
+                return i + 1
+
+            loop(0)
+
+            qml.CNOT(wires=[0, 1])
+
+        jaxpr = jax.make_jaxpr(circuit)()
+
+        assert len(jaxpr.eqns) == 3
+
+        assert jaxpr.eqns[0].primitive == qml.CNOT._primitive
+        assert jaxpr.eqns[1].primitive == while_loop_prim
+        assert jaxpr.eqns[2].primitive == qml.CNOT._primitive
+
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts)
+        jaxpr_ops = collector.state["ops"]
+        assert len(jaxpr_ops) == 5
+
+        expected_ops = [
+            qml.CNOT(wires=[0, 1]),
+            qml.Rot(np.pi / 2, np.pi, np.pi / 2, wires=[0]),  # RX and RZ fused
+            qml.Rot(np.pi / 2, np.pi, np.pi / 2, wires=[0]),  # RX and RZ fused
+            qml.Rot(np.pi / 2, np.pi, np.pi / 2, wires=[0]),  # RX and RZ fused
+            qml.CNOT(wires=[0, 1]),
+        ]
+
+        for op1, op2 in zip(jaxpr_ops, expected_ops):
+            # The qml.equal function does not recognize two qml.Rot operators
+            # as equivalent unless the input angles are exactly the same
+            assert op1.name == op2.name
+            assert qml.math.allclose(op1.parameters, op2.parameters)
+            assert op1.wires == op2.wires
+
+
+class TestSingleQubitFusionPLXPR:
+    """Unit tests for the single-qubit fusion transformation on PLXPRs."""
+
+    @pytest.mark.parametrize(
+        "atol, exclude_gates",
+        [(1e-5, None), (0.0, ["RY"])],
```

Comment:
Whether or not those `kwargs` are being used is not detected in the test, currently, right?
I think this is not dramatic, but if we don't test for it, we might as well not parametrize it :)